.github/workflows/ci-unit-tests.yml

```diff
@@ -0,0 +1,73 @@
+name: Unit Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```

Comment:
This feels rather dangerous given that the last step is deploy to Maven - what happens if we were to have more than one JDK versions in the matrix? Perhaps there should be another flag in the matrix controlling the deploy.

Reply:
Yeah, right now I'm only testing with 1.8, but it wasn't clear to me if multiple versions are being tested in Travis. Is there a set of versions we'd like to test against? If not, then we can probably just remove the matrix and things are fine. Otherwise, I think it's probably best to control the deploy via the matrix like you mention.